File: src/main/java/me/centralhardware/znatoki/telegram/country/days/tracker/ClickhouseConfiguration.java
package me.centralhardware.znatoki.telegram.country.days.tracker;

import com.clickhouse.jdbc.ClickHouseDataSource;
import org.apache.ibatis.mapping.Environment;
import org.apache.ibatis.session.SqlSessionFactory;
import org.apache.ibatis.session.SqlSessionFactoryBuilder;
import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory;

import javax.sql.DataSource;
import java.sql.SQLException;

public class ClickhouseConfiguration {

    public static DataSource getDataSource(){
        try {
            return new ClickHouseDataSource(System.getenv("CLICKHOUSE_URL"));
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static SqlSessionFactory getSqlSessionClickhouse(){
        var clickhouse = new Environment("clickhouse", new JdbcTransactionFactory(), getDataSource());
        var configurationClickhouse = new org.apache.ibatis.session.Configuration(clickhouse);

        configurationClickhouse.addMapper(CountryDaysTrackerMapper.class);
        configurationClickhouse.setCacheEnabled(false);

        return new SqlSessionFactoryBuilder().build(configurationClickhouse);
    }

}

File: src/main/kotlin/ClickhouseConfiguration.kt
import com.clickhouse.jdbc.ClickHouseDataSource
import org.apache.ibatis.mapping.Environment
import org.apache.ibatis.session.SqlSessionFactory
import org.apache.ibatis.session.SqlSessionFactoryBuilder
import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory
import java.sql.SQLException
import javax.sql.DataSource

object ClickhouseConfiguration {

    val dataSource: DataSource
        get() = try {
            ClickHouseDataSource(System.getenv("CLICKHOUSE_URL"))
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }

    val sqlSessionClickhouse: SqlSessionFactory
        get() {
            val clickhouse = Environment("clickhouse", JdbcTransactionFactory(), dataSource)
            val configurationClickhouse = org.apache.ibatis.session.Configuration(clickhouse)

            configurationClickhouse.addMapper(CountryDaysTrackerMapper::class.java)
            configurationClickhouse.isCacheEnabled = false

            return SqlSessionFactoryBuilder().build(configurationClickhouse)
        }
}